llmfoundry/callbacks/inactivity_callback.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2024 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import os
+import signal
+import threading
+from typing import Optional
+
+from composer import Callback, Logger, State
+from composer.loggers import MosaicMLLogger
+
+from llmfoundry.utils.exceptions import RunTimeoutError
+
+log = logging.getLogger(__name__)
+
+
+def _timeout(timeout: int, mosaicml_logger: Optional[MosaicMLLogger] = None):
+    log.error(
+        f'Timeout after no Trainer events were triggered for {timeout} seconds.',
+    )
+    if mosaicml_logger is not None:
+        mosaicml_logger.log_exception(RunTimeoutError(timeout=timeout))
+    os.kill(os.getpid(), signal.SIGINT)
+
+
+class InactivityCallback(Callback):
+
+    def __init__(
+        self,
+        timeout: int = 1800,
+        mosaicml_logger: Optional[MosaicMLLogger] = None,
```

Comment:
instead of passing via constructor, can you acquire the mosaic logger on `Event.INIT`? It should be on `state.callbacks` at that point. That allows the callback construction to not need a special case.